contracts/src/NativeTokenDestination.sol

```diff
@@ -317,7 +367,7 @@ contract NativeTokenDestination is
         }
 
         // Mint the tokens to this contract address.
-        _mint(address(this), amount);
+        _mintNativeCoin(address(this), amount);
```

Comment:
I'm just wondering if we should emit something on line 366 above if we're returning early. I can imagine that might be a confusing code path to debug

Reply:
I think the `CollateralAdded` and `TokensWithdrawn` events emitted in `_withdraw` are sufficient for tracing it IMO